lib/internal/util/comparisons.js

```diff
@@ -173,6 +173,11 @@ function innerDeepEqual(val1, val2, strict, memos) {
     return false;
   }
 
+  // Handle WeakMap and WeakSet
+  if (val1Tag === '[object WeakMap]' || val1Tag === '[object WeakSet]') {
```

Comment:
A tag may be manipulated while using `util.types.isWeakMap()` and `isWeakSet()` is safe. Please use those methods instead.